setup.py

```diff
@@ -52,21 +52,13 @@ def list_modules(dirname):
 
         return module_names
 
-    ext_modules = [
-        Extension('falcon.' + ext, [path.join('falcon', ext + '.py')])
-        for ext in list_modules(path.join(MYDIR, 'falcon'))]
-
-    ext_modules += [
-        Extension('falcon.util.' + ext,
-                  [path.join('falcon', 'util', ext + '.py')])
-
-        for ext in list_modules(path.join(MYDIR, 'falcon', 'util'))]
-
-    ext_modules += [
-        Extension('falcon.routing.' + ext,
-                  [path.join('falcon', 'routing', ext + '.py')])
+    dirs = ['falcon', 'falcon.util', 'falcon.routing']
+    ext_modules = []
 
-        for ext in list_modules(path.join(MYDIR, 'falcon', 'routing'))]
+    for each in dirs:
+        ext_modules += [Extension(each + '.' + ext,
+                        [path.join(*each.split('.'), ext + '.py')])
```

Comment:
This looks good, just needs a little modification. Currently, running this fails with `SyntaxError: only named arguments may follow *expression`. One way you might work around this is by using the extend operator (`+`), e.g.:

```py
path.join(*(each.split('.') + [ext + '.py'])) 
```